flowmachine/flowmachine/core/query.py

```diff
@@ -165,8 +165,8 @@ def __format__(self, fmt=""):
     def __iter__(self):
         con = get_db().engine
         qur = self.get_query()
-        with con.begin():
-            self._query_object = con.execute(qur)
+        with con.begin() as trans:
+            self._query_object = trans.execute(qur)
```

Comment:
This was my main concern for nested transactions, but it could be I've not read it through right - now I think about it, `__iter__` will only be called on top-level queries, and not by sub-queries?